internal/secrets/file/provider.go

```diff
@@ -0,0 +1,200 @@
+package file
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"go.flipt.io/flipt/internal/config"
+	"go.flipt.io/flipt/internal/secrets"
+	"go.uber.org/zap"
+)
+
+func init() {
+	// Register file provider factory
+	secrets.RegisterProviderFactory("file", func(cfg *config.Config, logger *zap.Logger) (secrets.Provider, error) {
+		if cfg.Secrets.Providers.File == nil {
+			return nil, fmt.Errorf("file provider configuration not found")
+		}
+		return NewProvider(cfg.Secrets.Providers.File.BasePath, logger)
+	})
+}
+
+// Provider implements secrets.Provider for file-based storage.
+type Provider struct {
+	basePath string
+	logger   *zap.Logger
+}
+
+// NewProvider creates a new file-based secret provider.
+func NewProvider(basePath string, logger *zap.Logger) (*Provider, error) {
+	// Ensure base path exists
+	if err := os.MkdirAll(basePath, 0700); err != nil {
+		return nil, fmt.Errorf("creating base path: %w", err)
+	}
```

Comment:
`os.Stat` probably is a better way to check if the path exists. If the file system is read only it will fails